internal/common/utils.go

```diff
@@ -44,14 +53,52 @@ func UpdateOperatingState(name string, state string, lc logger.LoggingClient, dc
 	}
 }
 
-func SendEvent(event *dtos.Event, correlationID string, lc logger.LoggingClient, ec interfaces.EventClient) {
+func SendEvent(event *dtos.Event, correlationID string, dic *di.Container) {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	configuration := container.ConfigurationFrom(dic.Get)
 	ctx := context.WithValue(context.Background(), CorrelationHeader, correlationID)
 
-	req := requests.NewAddEventRequest(*event)
-	res, err := ec.Add(ctx, req)
-	if err != nil {
-		lc.Errorf("failed to push event to core-data: %s", err)
+	if configuration.MessageQueue.Enabled {
+		encoding := clients.ContentTypeJSON
+		mc := container.MessagingClientFrom(dic.Get)
+		for _, r := range event.Readings {
+			if r.ValueType == v2.ValueTypeBinary {
+				encoding = clients.ContentTypeCBOR
+				break
+			}
+		}
+
+		var err error
+		var bytes []byte
+		switch encoding {
+		case clients.ContentTypeJSON:
+			ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeJSON)
+			bytes, err = json.Marshal(event)
+			if err != nil {
+				lc.Errorf("failed to encode event to JSON: %v", err)
+			}
+		case clients.ContentTypeCBOR:
+			ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeCBOR)
+			bytes, err = cbor.Marshal(event)
+			if err != nil {
+				lc.Errorf("failed to encode event to CBOR: %v", err)
+			}
+		}
+		env := types.NewMessageEnvelope(bytes, ctx)
+		err = mc.Publish(env, configuration.MessageQueue.PublishTopicPrefix)
+		if err != nil {
+			lc.Errorf("failed to publish event to message client: %v", err)
+		}
+
+		lc.Debugf("Event %s published to MessageBus", event.Id)
```

Comment:
Might be helpful to include `profilename`, `devicename` & `sourcename` in this message.